src/state/slices/tradeQuoteSlice/constants.ts

```diff
@@ -13,12 +13,18 @@ export const initialTransactionState = {
   state: TransactionExecutionState.AwaitingConfirmation,
 }
 
+const initialProgressState = {
```

Comment:
Note, this is not exported as this is not shared with the limit slice

Reply:
Also, the second guts of this, stores the progress of a hop in the tradeQuoteSlice since leveraging `checkTradeStatus` would be impossible at worst, dangerous at best because of the assumption that we would keep listeners on old references.

`checkTradeStatus` always overrides things and doesn't keep track of previous status, so figured out the best way would be to simply keep it in store - also makes things a lot easier to consume.